### ISSUE TITLE
Logic for pending transaction hyperledger besu

### ISSUE THREAD

**Issue by helderjnpinto** [blockscout/blockscout#2858]
I'm trying to change pending transactions for Hyperledger besu blockchain, so i change the endpoint of txpool but because Besu only sends txHash i need to fetch getTransactionByHash for each pending transaction, and set blockchash to 0x0 and blockNumber to nil, because Besu only give this info when its mined in IBFT PoA consensus. But in the web this transaction is not showing. 

I thing the logic of pending tx is explorer try to validate this transactions, but i check the database transactions table in the hash for pending tx is not there!

Below is the code for fetch details of pending transactions in besu ibft.
With `txpool_besuTransactions` querie only gets transaction hash, and for more detail needs to `eth_getTransactionByHash`.

`
 def fetch_pending_transactions(json_rpc_named_arguments) do

    with {:ok, transaction_data} <-
      %{id: 1, method: "txpool_besuTransactions", params: []} |> request() |> json_rpc(json_rpc_named_arguments) do

   

      transactions_params =
        transaction_data
        |>  Enum.map(fn( %{ "hash" => hash }) ->
                      %{id: 1, method: "eth_getTransactionByHash", params: [ hash ] } |> request() |> json_rpc(json_rpc_named_arguments) |> elem(1)
                     end)
        |> Enum.map(fn params ->
                      Map.merge(params, %{ "blockHash" => "0x0", "blockNumber" => "0x0", "transactionIndex" => "0x0" })
                    end)
        |> Transactions.to_elixir()
        |> Transactions.elixir_to_params()
        |> Enum.map(fn params ->
                      %{ params | block_hash: nil, index: nil }
                    end)

      {:ok, transactions_params}
    end
  end
`

### Environment

* Elixir & Erlang/OTP versions (`elixir -version`): 
Erlang/OTP 22 [erts-10.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]
Elixir 1.9.1 (compiled with Erlang/OTP 22)

* Operating System: ubuntu 18
* Blockscout Version/branch: latest

### Steps to reproduce
Besu is ethereum client for enterprise. All queries are compatible like geth.
.env 
```
COIN=USD
NETWORK=NEWCOIN
ETHEREUM_JSONRPC_VARIANT=geth 

ETHEREUM_JSONRPC_HTTP_URL=http://localhost:8545
ETHEREUM_JSONRPC_WS_URL=ws://localhost:8546
ETHEREUM_JSONRPC_TRACE_URL=http://localhost:8545
PORT=4000
```

Run node besu and send an transaction with very low gas price... this tx will be pending for testing. 
  
### Expected behaviour

Show pending transactions on web

### Actual behaviour

The pending transactions not show.

**Comment by vbaranov** [comment]
@helderjnpinto I am curious is Hyperledger Besu `eth_getTransactionByHash` JSON RPC method actually return details for tx from txpool?

**Comment by Matias-Gallardo** [comment]
I've tried this code but I always get an error when compiling.

**Comment by helderjnpinto** [comment]
> @helderjnpinto I am curious is Hyperledger Besu `eth_getTransactionByHash` JSON RPC method actually return details for tx from txpool?

![image](https://user-images.githubusercontent.com/12281088/83511180-2c2f5200-a4c6-11ea-9496-c559e9ea0ed4.png)

**Comment by vbaranov** [comment]
@helderjnpinto but this tx looks like not pending transaction since it is already included to the block (`blockNumber` is returned inn the response)

**Comment by helderjnpinto** [comment]
@vbaranov rpc for pending transactions only have the transaction hash this is not the payload for that.

**Comment by helderjnpinto** [comment]
@vbaranov https://besu.hyperledger.org/en/stable/Reference/API-Methods/#txpool_besutransactions
![image](https://user-images.githubusercontent.com/12281088/83540161-d66f9f80-a4ef-11ea-9808-2c5dc9175b1a.png)

**Comment by vbaranov** [comment]
@helderjnpinto yes, I know. And this is why you tried to get pending tx details from `eth_getTransactionByHash` by pending transaction hash. But does `eth_getTransactionByHash` return data for pending tx hash? There is no information in Besu docs about that. At least, I didn't find. And in the screenshot, you posted previously, https://github.com/poanetwork/blockscout/issues/2858#issuecomment-637453319, I suppose, it was not a pending tx hash since response contains `blockNumber`

**Comment by helderjnpinto** [comment]
@vbaranov yes sorry, the previously screen shot is a mined transaction, but besu give some information with this getTransactionByHash for pending tx, but blockNumber, blockHash, TransactionIndex not return, for this reason i put in 0x0 like in the first example

... ` Map.merge(params, %{ "blockHash" => "0x0", "blockNumber" => "0x0", "transactionIndex" => "0x0" })` . 

But there is a lot of payloads not equal to GETH, like in TRACE api, i get a lot of errors. Maybe the besu needs to change the payloads for each query to work like GETH for this.  

TRACE e.g 
![image](https://user-images.githubusercontent.com/12281088/83543981-22711300-a4f5-11ea-9ea8-39a7f8cb3e69.png)

if i fix this values `:erlang.binary_to_integer("0xa0", 16)` to :erlang.binary_to_integer("a0", 16) others will rising ... 

Transaction inputs payloads like "0x000000000x", transaction hash invalids etc... 

 I don't know where to change anymore :D

**Comment by vbaranov** [comment]
@helderjnpinto thanks a lot for the thorough answer. I see now that it is possible to get pending tx details.
